base_layer/p2p/src/services/liveness/service.rs

```diff
@@ -178,11 +178,17 @@ where
         } = msg;
         let node_id = source_peer.node_id;
         let public_key = source_peer.public_key;
+        let msg_useragent = ping_pong_msg.useragent.clone();
```

Comment:
Save a clone: Remove this and use the message directly in logs. In a few other places too - I don't think (could be wrong) that you'd need to clone the incoming user agent string at all

Reply:
You are correct, updated.